tests/units/plugins/filter/test_merge.py

```diff
@@ -0,0 +1,258 @@
+"""
+Tests for the "merge" plugin filter
+"""
+from copy import deepcopy
+
+from ansible.compat.tests import unittest
+from ansible.errors import AnsibleFilterError
+
+from plugins.filter.merge import merge_with_app
+
+
+def deep_sort_dict(d):
+    """
+        Recursively alphabetically sort lists as dictionary values (input dict
+        is not modified)
```

Comment:
I think it is modified :thinking: 
You could start from an empty defaultdict and fill it recursively with what is returned by the next call?